radis/lbl/base.py

```diff
@@ -1442,17 +1451,67 @@ def _calc_noneq_parameters(self, vib_distribution, singleTvibmode):
         the nonequilibrium energies)
         """
 
+        # Load variables
+        molecule = self.input.molecule
+        isotope = self.input.isotope
+        state = "X"
+
+        df = self.df0
+        if len(df) == 0:
+            return  # no lines
+
+        # Initializing a cache file for lookup later
+        filename = (
+            splitext(self.params.dbpath.split(",", 1)[0])[0].rsplit("/", 1)[0]
+            + "/ORIGINAL_FILE_new_columns"
+        )
+        if cache:
+            for isotope in self._get_isotope_list(molecule):
+
+                elec_state = self.get_partition_function_calculator(
+                    molecule, isotope, state
+                ).ElecState
+                fname = elec_state.jsonfile
+                # f1 = splitext(self.params.dbpath.split(",",1)[0])[0].rsplit("/",1)[0] + "/{0}".format(cache_file)
+                fcache = DataFileManager(engine).cache_file(filename)
+                # Generate cache file for later use
+                if cache:
+                    new_metadata = {
+                        # Last modification time of the original file :
+                        "last_modification": time.ctime(getmtime(fname)),
+                        "spectroscopic_constant_file": fname,
+                        "df_last_index": df.index[-1],
+                    }
+                    if self.verbose:
+                        print(
+                            "Generating cache file {0} with metadata :\n{1}".format(
+                                fcache, new_metadata
+                            )
+                        )
+                    try:
+                        save_to_hdf(
+                            df,
```

Comment:
Warning @sagarchotalia. You shouldn't be caching the Dataframe.
Cache the non energies columns only. And not in this fname, but in an _extra column alongside the line database files

Reply:
Evib, Erot etc is fetched only line 1537+.

Caching must happen after that.